tests/foreman/cli/test_import.py

```diff
@@ -694,6 +788,196 @@ def test_import_host_collections_recovery(self, test_data):
+            )
+        # import the prerequisities
+        import_org = Import.organization_with_tr_data(
+            {'csv-file': files['users']}
+        )
+        # now proceed with importing the repositories
+        import_repo = Import.repository_with_tr_data(
+            {
+                'csv-file': files['repositories'],
+                'synchronize': True,
+                'wait': True,
+            }
+        )
+        for result in (import_org, import_repo):
+            self.assertEqual(result[0].return_code, 0)
+
+        # get the sat6 mapping of the imported organizations
+        imp_orgs = get_sat6_id(csv_to_dataset([files['users']]), import_org[1])
+        repos_before = []
+        for imp_org in imp_orgs:
+            repos_before.append(
+                Repository.list(
+                    {'organization-id': imp_org['sat6']}
+                ).stdout
+            )
+        # Reimport the same repos and check for changes in sat6
+        self.assertEqual(
+            Import.repository(
+                {
+                    'csv-file': files['repositories'],
+                    'synchronize': True,
+                    'wait': True,
+                }
+            ).return_code, 0
+        )
+        self.assertEqual(
+            repos_before,
+            [
+                Repository.list({'organization-id': imp_org['sat6']}).stdout
+                for imp_org in imp_orgs
+            ]
+        )
+
+    @data(*import_repo_data())
+    def test_import_repo_recovery(self, test_data):
+        """@test: Try to Import Repos with the same name to invoke
+        usage of a recovery strategy (rename, map, none)
+
+        @feature: Import Repository Recover
+
+        @assert: 2nd Import will rename the new repos, 3rd import will
+        map them and the 4th one will result in No Action Taken
+
+        """
+        # prepare the data
+        files = dict(self.default_dataset[1])
+        # randomize the values for orgs and repos
+        files = dict(self.default_dataset[1])
+        for file_ in zip(
+            ['users', 'repositories'],
+            [u'organization_id', u'org_id'],
+        ):
+            files[file_[0]] = update_csv_values(
+                files[file_[0]],
+                file_[1],
+                test_data[file_[0]],
+                self.default_dataset[0]
+            )
+        # import the prerequisities
+        import_org = Import.organization_with_tr_data(
+            {'csv-file': files['users']}
+        )
+        for result in (
+            import_org,
+            Import.repository_with_tr_data(
+                {'csv-file': files['repositories']}
+            ),
+        ):
+            self.assertEqual(result[0].return_code, 0)
+        # clear the .transition_data to clear the transition mapping
+        ssh.command('rm -rf ${HOME}/.transition_data/repositories*')
+        ssh.command('rm -rf ${HOME}/.transition_data/products*')
```

Comment:
Please surround the `${HOME}` variable with double quotes. This will probably never cause an error. Probably. But there's a huge swath of issues that crop up with variable expansions. A mundane issue is with whitespace expansion:

```
[ichimonji10@beech:tmp]$ mkdir 'tv shows'
[ichimonji10@beech:tmp]$ rmdir tv shows
rmdir: failed to remove ‘tv’: No such file or directory
rmdir: failed to remove ‘shows’: No such file or directory
```

A more subtle issue is when values are gobbled up by utilities:

```
[ichimonji10@beech:tv shows]$ mkdir -- '-n FOO'
[ichimonji10@beech:tv shows]$ l -1
Code Geass: Lelouch of the Rebellion
Cowboy Bebop
Full Metal Alchemist: Brotherhood
-n FOO
Rurouni Kenshin: Trust & Betrayal
[ichimonji10@beech:tv shows]$ for dir in *; do echo $dir; done
Code Geass: Lelouch of the Rebellion
Cowboy Bebop
Full Metal Alchemist: Brotherhood
FOORurouni Kenshin: Trust & Betrayal
```

Just for fun, try to make this work sometime:

```sh
for dir in $(ls .); do echo $dir; done
```